python/xorbits/_mars/tensor/fuse/jax.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2022-2023 XProbe Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+
+try:
+    import jax
+
+    JAX_INSTALLED = True
+except ImportError:
+    JAX_INSTALLED = False
+
+from .. import arithmetic, reduction
+from ..array_utils import as_same_device
+from ..operands import TensorFuse
+from .core import TensorFuseChunkMixin
+
+
+class TensorJAXFuseChunk(TensorFuse, TensorFuseChunkMixin):
+    _op_type_ = None  # no opcode, cannot be serialized
+
+    @classmethod
+    def execute(cls, ctx, op):
+        chunk = op.outputs[0]
+        inputs = as_same_device([ctx[c.key] for c in op.inputs], device=op.device)
+        input_dict = {c.key: i for c, i in zip(op.inputs, inputs)}
+        res = np.array(_evaluate(chunk, input_dict))
+        res = _maybe_keepdims(chunk, res)
+        if chunk.ndim == 0 and res.ndim == 1 and res.size == 0:
+            res = res.dtype.type(0)
+        ctx[chunk.key] = res
+
+
+# recursively compute the inputs
+def _evaluate(chunk, input_dict):
```

Comment:
I benchmarked this function with the following test and found it performed 12.5% better than numexpr!!!

```python
def test_base_execution(setup):
    rs = np.random.RandomState(0)
    raw1 = rs.randint(10, size=(1000, 1000, 1000))
    raw2 = rs.randint(10, size=(1000, 1000, 1000))
    arr1 = tensor(raw1, chunk_size=50)
    arr2 = tensor(raw2, chunk_size=50)
    arr3 = arr1 + arr2 + 10
    res3 = arr3.execute().fetch()
```

I'm thinking if it is possible to generate a function like what we do with numexpr, so that we can get rid of the `if` statements and apply jit.

What do you think?

Reply:
yeah i can try doing that

Reply:
We've decided to optimize this function later.